rs/rosetta-api/tests/system_tests/common/system_test_environment.rs

```diff
@@ -0,0 +1,126 @@
+use crate::common::{
+    constants::{DEFAULT_INITIAL_BALANCE, STARTING_CYCLES_PER_CANISTER},
+    utils::test_identity,
+};
+use candid::{Encode, Principal};
+use ic_agent::Identity;
+use ic_icp_rosetta_client::RosettaClient;
+use ic_icp_rosetta_runner::{start_rosetta, RosettaContext, RosettaOptionsBuilder};
+use ic_icrc1_test_utils::minter_identity;
+use ic_ledger_test_utils::build_ledger_wasm;
+use ic_ledger_test_utils::pocket_ic_helpers::ledger::LEDGER_CANISTER_ID;
+use ic_rosetta_test_utils::path_from_env;
+use ic_types::PrincipalId;
+use icp_ledger::{AccountIdentifier, LedgerCanisterInitPayload};
+use pocket_ic::{nonblocking::PocketIc, PocketIcBuilder};
+use std::collections::HashMap;
+use tempfile::TempDir;
+
+pub struct RosettaTestingEnvironment {
+    pub _pocket_ic: PocketIc,
+    pub _rosetta_context: RosettaContext,
+    pub rosetta_client: RosettaClient,
+}
+
+impl RosettaTestingEnvironment {
+    pub fn builder() -> RosettaTestingEnviornmentBuilder {
+        RosettaTestingEnviornmentBuilder::new()
+    }
+}
+
+pub struct RosettaTestingEnviornmentBuilder {
+    pub persistent_storage: bool,
+    pub icp_ledger_init_args: Option<LedgerCanisterInitPayload>,
```

Comment:
Do these need to be public?

Reply:
We generally always keep fields public unless there is a good reason to not do so. This coding practice was introduced by Roman a while back.

Reply:
I think it is a bad practice!